### ISSUE TITLE
missing ffi.h

### ISSUE THREAD

**Issue by mvhulten** [ReneVolution/profanity-omemo-plugin#16]
During `setup.py` I get:

    removing: _configtest.c _configtest.o
    c/_cffi_backend.c:15:17: fatal error: ffi.h: Bestand of map bestaat niet
     #include <ffi.h>
                     ^
    compilation terminated.

I guessed that I needed to install `cffi`.  I am using Debian GNU/Linux.  Should I install with `pip` or through APT?  `pip` tries to write to `/usr/`; whereas the Debian package is maybe not picked up, it seems.

**Comment by ReneVolution** [comment]
This https://cryptography.io/en/latest/installation/#building-cryptography-on-linux seems relevant. I don't have so if you could test it, i'd add a hint in the README.

**Comment by bascht** [comment]
@mvhulten I think you're missing the cffi bindings. I think you're safe to use the ones that come with Debian. In your case [`python-cffi`](https://packages.debian.org/jessie/python-cffi).

If you have more missing dependencies, see also: https://github.com/bascht/profanity-omemo-plugin/blob/master/tests/integration/Dockerfile.profanity#L21 :)

**Comment by mvhulten** [comment]
The Debian packages suggested by @bascht were needed but not sufficient.  I had to follow @ReneVolution's suggestion.  It was crucial that I install `libffi-dev`.

Additionally, there were two more actions I needed to do before installing  cryptography through PyPI.
- remove the Debian `python-cryptography` and `python-protobuf` packages;
- upgrade of all PyPI packages (don't know which one(s) was/were essential).

More than one package manager (in this case Debian and PyPI) often gives me headaches.  In this case I *need* `libffi-dev` from Debian and I *must not have* two Debian python packages.
